onnxruntime/python/tools/onnxruntime_test.py

```diff
@@ -80,39 +50,78 @@ def main():
+
+    sess = onnxrt.InferenceSession(model_path, sess_options)
+    meta = sess.get_modelmeta()
+
+    if not feeds:
+        feeds = generate_feeds(sess, symbolic_dims)
+
+    if override_initializers:
+        # Starting with IR4 some initializers provide default values
+        # and can be overridden (available in IR4). For IR < 4 models
+        # the list would be empty
+        for initializer in sess.get_overridable_initializers():
+            shape = [dim if dim else 1 for dim in initializer.shape]
+            if initializer.type in float_dict:
+                feeds[initializer.name] = np.random.rand(*shape).astype(float_dict[initializer.type])
+            elif initializer.type in integer_dict:
+                feeds[initializer.name] = np.random.uniform(high=1000,
+                                                            size=tuple(shape)).astype(integer_dict[initializer.type])
+            elif initializer.type == 'tensor(bool)':
+                feeds[initializer.name] = np.random.randint(2, size=tuple(shape)).astype('bool')
+            else:
+                print("unsupported initializer type {} for initializer {}".format(initializer.type, initializer.name))
+                sys.exit(-1)
 
     start = timer()
-    for i in range(iters):
-        sess.run([], feeds)  # fetch all outputs
+    for i in range(num_iters):
+        outputs = sess.run([], feeds)  # fetch all outputs
     end = timer()
 
     print("model: {}".format(meta.graph_name))
     print("version: {}".format(meta.version))
-    print("iterations: {}".format(iters))
-    print("avg latency: {} ms".format(((end - start) * 1000) / iters))
+    print("iterations: {}".format(num_iters))
+    print("avg latency: {} ms".format(((end - start) * 1000) / num_iters))
 
-    if args.profile:
+    if profile:
         trace_file = sess.end_profiling()
         print("trace file written to: {}".format(trace_file))
 
-    return 0
+    return 0, feeds, num_iters > 0 and outputs
 
 
 if __name__ == "__main__":
-    sys.exit(main())
+    parser = argparse.ArgumentParser(description='Simple ONNX Runtime Test Tool.')
+    parser.add_argument('model_path', help='model path')
+    parser.add_argument('num_iters', nargs='?', type=int, default=1000, help='model run iterations. default=1000')
+    parser.add_argument('--debug', action='store_true', help='pause execution to allow attaching a debugger.')
+    parser.add_argument('--profile', action='store_true', help='enable chrome timeline trace profiling.')
+    parser.add_argument('--symbolic_dims', default={}, type=lambda s: dict(x.split("=") for x in s.split(",")),
+                        help='Comma separated name=value pairs for any symbolic dimensions in the model input. '
+                             'e.g. --symbolic_dims batch=1,seqlen=5. '
+                             'If not provided, the value of 1 will be used for all symbolic dimensions.')
+
+    args = parser.parse_args()
```

Comment:
>args [](start = 4, length = 4)

Could we add an option to compare the runs between EP vs. CPU in this file?  That would save you the model_tools.py for verifying results.

Reply:
It would be helpful that onnxruntime_test has an option to compare runs between EPs. However, what model_tools does is specific to model_editor - it compares runs between original and edited models. So I think logic in model_tools shall stay.